src/main/java/io/gatling/plugin/EnterprisePluginClient.java

```diff
@@ -33,9 +33,16 @@ public EnterprisePluginClient(EnterpriseClient enterpriseClient) {
   }
 
   @Override
-  public long uploadPackage(UUID packageId, File file) throws EnterpriseClientException {
-    nonNullParam(packageId, "packageId");
+  public long uploadPackage(UUID packageId, UUID simulationId, File file)
+      throws EnterpriseClientException {
     nonNullParam(file, "file");
+    if (packageId == null) {
+      if (simulationId == null) {
+        throw new IllegalArgumentException("You need to set either packageId or simulationId");
+      }
+      Simulation simulation = enterpriseClient.getSimulation(simulationId);
+      return enterpriseClient.uploadPackageWithChecksum(simulation.pkgId, file);
+    }
     return enterpriseClient.uploadPackageWithChecksum(packageId, file);
   }
```

Comment:
```suggestion
  public long uploadPackage(UUID packageId, UUID simulationId, File file)
      throws EnterpriseClientException {
    nonNullParam(file, "file");
    
    if (packageId == null && simulationId == null) {
      throw new IllegalArgumentException("You need to set either packageId or simulationId");
    }
    
    if (packageId != null) {
      return enterpriseClient.uploadPackageWithChecksum(packageId, file);
    } else {
      Simulation simulation = enterpriseClient.getSimulation(simulationId);
      return enterpriseClient.uploadPackageWithChecksum(simulation.pkgId, file);
    }
  }
```

Reply:
I would rather split the method in two, so that plugin can throw a proper error (with configuration) when both are not configured.

```suggestion
  public long uploadPackage(UUID packageId, File file)
      throws EnterpriseClientException {
    nonNullParam(packageId, "packageId");
    nonNullParam(file, "file");
    
    return enterpriseClient.uploadPackageWithChecksum(packageId, file);
  }
  
  public long uploadPackageToSimulation(UUID simulationId, File file) throws EnterpriseClientException {
     nonNullParam(packageId, "packageId");
     nonNullParam(file, "file");
  
     Simulation simulation = enterpriseClient.getSimulation(simulationId);
     return enterpriseClient.uploadPackageWithChecksum(simulation.pkgId, file);
  }
```